src/frontend/src/app/AppPublic.tsx

```diff
@@ -35,10 +35,20 @@ const AppPublic: React.FC = () => {
     }
 
     // otherwise, the user needs to login manually
+    // prepare query args to store path after login
+    const pathParts: string[] = history.location.pathname.split('/').slice(1);
+    // if the URL ended in an extra '/' we don't want to clutter the query args
+    if (pathParts[pathParts.length - 1] === '') pathParts.pop();
```

Comment:
shouldn't this be === '/' ? or am i misreading

Reply:
`'/a/b/c/'.split('/')` gives you `['', 'a', 'b', 'c', '']`